test/token/governance/ERC20Base.t.sol

```diff
@@ -17,6 +17,31 @@ abstract contract ERC20AddressHelper {
     address public signer = address(0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266);
 }
 
+contract ERC20BaseInitializationTest is ERC20AddressHelper, Test {
+    ERC20Base public impl;
+    ProxyAdmin public proxyAdmin;
+    ERC20Base public token;
+
+    function testInitializing() public {
+        vm.startPrank(deployer);
+        impl = new ERC20Base();
+        proxyAdmin = new ProxyAdmin();
+
+        TransparentUpgradeableProxy proxy;
+        proxy = new TransparentUpgradeableProxy(
+            address(impl),
+            address(proxyAdmin),
+            ""
+        );
+        token = ERC20Base(address(proxy));
+        vm.expectRevert("ERC20Base: Token name cannot be empty");
+        token.initialize(owner, "", "", 10000000000000000000000000000);
```

Comment:
If this derived from `ERC20Basehelper` and used the `deployNewToken` function, you'd be able to make the same assertions in a bit more concise manner. The above would become:

```suggestion
        vm.startPrank(deployer);
        vm.expectRevert("ERC20Base: Token name cannot be empty");
       ERC20Base newToken = deployNewToken(owner, "", "", 10000000000000000000000000000);
```

Reply:
I tried to do this but could never get the call to deployNewToken to run again. It would run as part of the constructor from BaseHelper but then would not execute again. It appeared to run "upgrades?". I reverted and fixed the first issue and figured we can work on this together if the change is needed.

Reply:
This sounds like an issue with foundry's spies not being able to instrument the calls that are made from inside `deployNewToken`. No need to make any changes here since what you had already worked.